casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -173,22 +173,24 @@ class RuntimeManagerImpl[F[_]: Concurrent] private[rholang] (
   private def computeDeployPayment(
       runtime: Runtime[F]
   )(user: ByteString, amount: Long): F[Checkpoint] =
-    computeEffect(runtime, runtime.reducer)(
-      ConstructDeploy.sourceDeployNow(deployPaymentSource(amount)).withDeployer(user)
-    ).ensure(BugFoundError("Deploy payment failed unexpectedly"))(_.errors.isEmpty) >>
-      getResult(runtime)() >>= {
-      case Seq(RhoType.Tuple2(RhoType.Boolean(true), Par.defaultInstance)) =>
+    for {
+      _ <- computeEffect(runtime, runtime.reducer)(
+            ConstructDeploy.sourceDeployNow(deployPaymentSource(amount)).withDeployer(user)
+          ).ensure(BugFoundError("Deploy payment failed unexpectedly"))(_.errors.isEmpty)
+      result <- getResult(runtime)() >>= {
```

Comment:
since its already in a for, drop the flatmap?

Reply:
would rather not, this avoids the awkward formatting that would happen due to `result match { //...`

Reply:
Nah, the formatting is just as awkward :) Done :)